commands/http/parse.go

```diff
@@ -20,12 +20,20 @@ func Parse(r *http.Request, root *cmds.Command) (cmds.Request, error) {
 
 	stringArgs := make([]string, 0)
 
+	if err := apiVersionMatches(r); err != nil {
```

Comment:
the check should not be in the `commands` lib, as the commands lib is not-ipfs specific and will be ripped out into its own lib.

it should be in ipfs specific stuff. if no function called globally for all daemon commands then we should add one that could run before the command is invoked.